2020/09/02/rclone-upload-to-cloud-from-cli/rclone-upload-to-cloud-from-cli.html.md

```diff
@@ -0,0 +1,240 @@
+﻿---
+author: "Ardyn Elisabeth Majere"
+title: "Rclone - upload to the cloud from your command line- And much more."
+tags: cli, linux, cloud, storage,
+---
+
+### The swiss army knife of storage.
+
+Oftentimes, cloud storage providers like Google Drive are great solutions for storing files- You can upload the data and not have to worry about maintaining a separate system to host it, with all the security hassles that can bring. However- very few of the major cloud storage providers offer a command line interface, or any official way to upload without using their web interface or closed-source binary tools, if that.
+
+This obviously makes uploading files from servers difficult- but not impossible, if you know the right tools.
+
+Previously, Jon Jensen penned a blog post about [gdrive](https://www.endpoint.com/blog/2019/09/30/google-drive-for-vm-images), a Google Drive command line tool. However, due to changes with Google's Drive security, that tool no longer works. This led me to look for a replacement.
+
+### Our use case
+
+Recently I had to put some large files in to long term storage on Google Drive, as we needed the space back. We wanted to retain the data, but didn't foresee needing to access it for some time, if ever. Google Drive was a good solution for us, however, the problem became how to get it there.
+
+They were too big, and some of them were not stored sparsely- empty space was tacked on to the disk images. There was also the issue of data integrity as the drives potentially contained customer information, so we wanted to encrypt them. There was also the issue of sequentially processing the files- encrypting them- then uploading them- I felt this would take quite a bit of time.
```

Comment:
Does "sparsely" here just mean that they weren't efficiently stored? Or does it refer to a [sparse matrix](https://en.wikipedia.org/wiki/Sparse_matrix)?

Reply:
@sethjensen1 Unix sparse file storage: https://en.wikipedia.org/wiki/Sparse_file

Reply:
👍 I had a sneaking suspicion this referred to a storage method.